### PR TITLE
Add extensions

### PR DESCRIPTION
**Pull request by J4m3s00**
- [x] Add hot reloading (Keep ImGui in min)
- [x] Add custom panels from extension

### CODE REVIEW

File: src/ui/engine_panels.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+namespace Engine {
+
+    struct ComponentDescription
+    {
```
Comment (J4m3s00) [inline src/ui/engine_panels.h:6]: Add some kind of user data buffer

File: src/ui/engine_ui.cpp [hunk 2]
```diff
@@ -120,197 +116,19 @@ void UI::Render()
...
-
-void UI::RenderInputField(const EString& label, EProperty<EUUID>& value) 
-{
-    ImGui::Text("%s", label.c_str());
-    ImGui::SameLine();
-    ImGui::Text("%s", value.GetValue().ToString().c_str());
-}
-
-void UI::RenderInputField(const EString& label, EObjectProperty<ETexture2D>& value) 
+void UI::ResetContext()
```
Comment (J4m3s00) [inline src/ui/engine_ui.cpp:119]: Check ich used

File: src/ui/engine_ui_functions.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    namespace UI {
+    bool IsItemActiveLastFrame();
+
+    static bool IsItemJustReleased()
+    {
+        return IsItemActiveLastFrame() && !ImGui::IsItemActive();
+    }
+
+
+    void RenderInputField_Bool(const EString& label, EProperty<bool>* value);
```
Comment (J4m3s00) [inline src/ui/engine_ui_functions.h:13]: I think we don't need these anymore

**Review by J4m3s00 — COMMENTED** [review]
Looks good

File: src/file/engine_folder.cpp [hunk 4]
```diff
@@ -2,14 +2,19 @@
...
 using namespace Engine;
 
-EFolder::EFolder(const EString& path) 
+EFolder::EFolder(const EString& path, i32 folderOptions) 
+    : fFolderOptions(folderOptions)
 {
     fFullPath = path;
+    if (fFolderOptions & EFolderOptions_CreateIfNotExist && !Exist())
+    {
+        Create();
```
Comment (valentinstoecker) [inline src/file/engine_folder.cpp:11]: test

Comment (J4m3s00) [reply src/file/engine_folder.cpp:11]: Nice

**Comment by valentinstoecker** [comment]
hallo

**Comment by valentinstoecker** [comment]
> Nice 

bajh

**Comment by valentinstoecker** [comment]
@J4m3s00 Hjhfjg

**Pull request merged**